### ISSUE TITLE
install fail with "undefined reference to 'MG_Uninitialize' "

### ISSUE THREAD

**Issue by aj3423** [NaniteFactory/gominhook#1]
Hi, I try to install this library with 

    $ go get -v github.com/nanitefactory/gominhook

But there are lots of error

	github.com/nanitefactory/gominhook
	# github.com/nanitefactory/gominhook
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x0): undefined reference to `MH_Uninitialize'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x4): undefined reference to `MH_RemoveHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x8): undefined reference to `MH_QueueEnableHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0xc): undefined reference to `MH_QueueDisableHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x10): undefined reference to `MH_Initialize'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x14): undefined reference to `MH_EnableHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x18): undefined reference to `MH_DisableHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x1c): undefined reference to `MH_CreateHookApiEx'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x20): undefined reference to `MH_CreateHookApi'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x24): undefined reference to `MH_CreateHook'
	c:\temp\go-build110183186\b001\_cgo_main.o:_cgo_main.c:(.data+0x28): undefined reference to `MH_ApplyQueued'
	collect2.exe: error: ld returned 1 exit status

I see these functions are exported from MinHook.xxx.dll, but how to solve it?

Thanks.

**Comment by NaniteFactory** [comment]
It seems that your build is failing. I'm not sure for this, but I just guess that it could be something specific to your environment. It might be the version of OS you have, or it could be the version of Go you're using, or your msys, etc. etc.

Could you give me your system details? I'll try to investigate further and get back to you.

Mine shows: 
```
C:\Users\nanit>go version
go version go1.10.2 windows/amd64

C:\Users\nanit>go env
set GOARCH=amd64
set GOBIN=D:\GoPath\bin
set GOCACHE=C:\Users\nanit\AppData\Local\go-build
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=D:\GoPath
set GORACE=
set GOROOT=D:\Go
set GOTMPDIR=
set GOTOOLDIR=D:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\nanit\AppData\Local\Temp\go-build902006178=/tmp/go-build -gno-record-gcc-switches
```

**Comment by aj3423** [comment]
I found that the installing requires cgo, so it requires gcc. After installing msys2 and gcc, it works now.

**Comment by NaniteFactory** [comment]
I'm not sure why you closed this. This is a valid issue. It actually didn't work for x86 mode, so I spent hours to reproduce the problem and finally figured it out. xD
`MinHook.x86.dll` wasn't just something GCC compatible, (probably because it was built with VC++) thus our linker cannot link it.
It might link fine using Microsoft's `link.exe`, but for this case I simply built another binary with MinGW.
I named it `MinHook.gcc.x86.dll`. It's in the binary folder of this repo so I believe everything should work fine now.
This issue is fixed by 5f465acb49c0986b8490a13566fb0fee98da5f46.
